### PR TITLE
Script Loader for vanilla js workloads

### PR DESCRIPTION
**Pull request by flashdesignory**
This pr adds an example to load the 'benchmark-connector.min.js' file at a later time during `requestIdleCallback`, to ensure we're sending `app-ready` postMessage when the app is really ready.

Why is this important?
Benchmarks need to know when the workload is ready to run a test on it. Previously, a clunky function waited for certain elements to be ready instead, which doesn't guarantee that everything is initialized. 
This way (this pr), the workload has the power to decide when it is really really ready.

Basically it's a super simple script loader, to give a simplified version of what frameworks offer out of the box 😄 

What's important to look at:
- ignore the dist folder
- look at the script-loader.js file

@kara

### CODE REVIEW

File: apps/todomvc/todomvc-web-components/dist/utils/script-loader.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+function buildScript({scriptEl, code}) {
+    return new Promise((resolve) => {
+        scriptEl.textContent = code;
+        return resolve({ success: true, type: "buildScript"});
+    });
+}
+
+export async function loadScript({ id, url = "", code = "", type = "", strategy = "default", location = "body", onError = () => {}, onSuccess = () => {} } = {}) {
```
Comment (camillobruni) [inline apps/todomvc/todomvc-web-components/dist/utils/script-loader.js:44]: onSuccess / onError seem a  bit strange given that this is an async function, doesn't the async function throw on failure?

File: apps/todomvc/todomvc-web-components/dist/utils/request-idle-callback.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            });
+        }, 1)
+
+        return timeoutId;
+    }
+
+export const cancelIdleCallback = window.cancelIdleCallback =
+    window.cancelIdleCallback ||
+    function (id) {
+        window.clearTimeout(id);
```
[the remark this answers is not in the document]
Comment (flashdesignory) [reply apps/todomvc/todomvc-web-components/dist/utils/request-idle-callback.js]: done - thanks 😄

File: apps/todomvc/todomvc-web-components/dist/utils/script-loader.js (hunk 1 above)
Comment (flashdesignory) [reply apps/todomvc/todomvc-web-components/dist/utils/script-loader.js]: reworked this.

File: apps/todomvc/todomvc-web-components/src/utils/request-idle-callback.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+export const requestIdleCallback = window.requestIdleCallback =
+    window.requestIdleCallback ||
+    function (cb) {
+        const start = Date.now();
+        const timeoutId = window.setTimeout(function () {
+            cb({
+                didTimeout: false,
+                timeRemaining: function () {
```
Comment (camillobruni) [inline apps/todomvc/todomvc-web-components/src/utils/request-idle-callback.js:8]: This could just be `timeRemaining() {`

File: apps/todomvc/todomvc-web-components/dist/utils/script-loader.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    await addScript({scriptEl, location});
+
+    if (code !== "") {
+        // add inline code to script element
+        await buildScript({scriptEl, code});
+        onSuccess?.();
+    } else {
+        // load external url
+        const {success} = await initScript({scriptEl, url, strategy});
+        success ? onSuccess?.() : onError?.();
```
Comment (camillobruni) [inline apps/todomvc/todomvc-web-components/dist/utils/script-loader.js:58]: I still find it a bit weird to mix the continuation-based approach (onXXX) with async-await here (but that might be just a matter of style).

I'd imagine something like this:
```
try {
  let scriptId = await loadScript(...);
  // onSuccess code goes here
} catch (error) {
  // onError code goes here
}
```

What't the advantage of passing back the `{success: true}` object?

File: apps/todomvc/todomvc-web-components/dist/utils/script-loader.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+function addScript({scriptEl, location}) {
+    return new Promise((resolve) => {
+        if (location === 'head') {
+            document.head.appendChild(scriptEl);
+        } else {
+            document.body.appendChild(scriptEl);
+        }
+        return resolve({ success: true, type: "addScript"});
+    });
+}
```
Comment (camillobruni) [inline apps/todomvc/todomvc-web-components/dist/utils/script-loader.js:20]: Are there plans to make this async at some point? If not, then I'm not sure we need the Promise here, since all the operations are sync (here and `createScript`, `buildScript`)

File: apps/todomvc/todomvc-web-components/dist/utils/script-loader.js (hunk 4 above)
Comment (flashdesignory) [reply apps/todomvc/todomvc-web-components/dist/utils/script-loader.js]: simplified and removed overhead that's not needed - lmk

File: apps/todomvc/todomvc-web-components/dist/utils/script-loader.js (hunk 5 above)
Comment (flashdesignory) [reply apps/todomvc/todomvc-web-components/dist/utils/script-loader.js]: reworked it and only kept promises for the loading

**Review by camillobruni — APPROVED** [review]
